Content.Server/_NF/Speech/EntitySystems/GoblinAccentSystem.cs

```diff
@@ -1,31 +1,45 @@
 using Content.Server._NF.Speech.Components;
 using Content.Server.Speech;
 using Content.Server.Speech.EntitySystems;
+using System.Text.RegularExpressions;
 
 namespace Content.Server._NF.Speech.EntitySystems;
 
+// The whole code is a copy of SouthernAccentSystem by UBlueberry (https://github.com/UBlueberry)
 public sealed class GoblinAccentSystem : EntitySystem
 {
+    private static readonly Regex RegexIng = new(@"ing\b");
+    private static readonly Regex RegexAnd = new(@"\band\b");
+    private static readonly Regex RegexEr = new(@"er\b");
+    private static readonly Regex RegexTt = new(@"tt");
+    private static readonly Regex RegexOf = new(@"\bof\b");
+    private static readonly Regex RegexThe = new(@"\bthe\b");
+    private static readonly Regex RegexH = new(@"\bh");
+    private static readonly Regex RegexSelf = new(@"self\b");
+
     [Dependency] private readonly ReplacementAccentSystem _replacement = default!;
 
     public override void Initialize()
     {
         base.Initialize();
-
-        SubscribeLocalEvent<GoblinAccentComponent, AccentGetEvent>(OnAccentGet);
+        SubscribeLocalEvent<GoblinAccentComponent, AccentGetEvent>(OnAccent);
     }
 
-    // converts left word when typed into the right word. For example typing you becomes ye.
-    public string Accentuate(string message, GoblinAccentComponent component)
+    private void OnAccent(EntityUid uid, GoblinAccentComponent component, AccentGetEvent args)
     {
-        var msg = message;
+        var message = args.Message;
 
-        msg = _replacement.ApplyReplacements(msg, "goblin");
-        return msg;
-    }
+        message = _replacement.ApplyReplacements(message, "goblin");
 
-    private void OnAccentGet(EntityUid uid, GoblinAccentComponent component, AccentGetEvent args)
-    {
-        args.Message = Accentuate(args.Message, component);
+        message = RegexIng.Replace(message, "in'");
+        message = RegexAnd.Replace(message, "an'");
+        message = RegexEr.Replace(message, "ah");
+        message = RegexTt.Replace(message, "'");
+        message = RegexH.Replace(message, "'");
+        message = RegexSelf.Replace(message, "sewf");
+        message = RegexOf.Replace(message, "o'");
+        message = RegexThe.Replace(message, "da");
+
```

Comment:
I've had trouble getting these to work as I'd expect.  For a few examples, typing and sending
"harpies", "of of of of", and "and and and and" don't seem to work.  Word replacement seems fine, and I'm not entirely sure why these regex checks don't work.

Maybe they're timing out?  Seems odd.

Reply:
The blasted thing is odd: if you rely on the innate accent of goblin species, it doesn't work for some reason, but if you tick "goblin cant" in traits, then it starts working. I wonder if there are any other species that have an innate accent trait. Maybe upstream has changed something about how it is defined in species? It might be that I'm missing something.
![image](https://github.com/new-frontiers-14/frontier-station-14/assets/65374927/aeec147f-cbb6-466a-ac42-43d9c273c020)

Reply:
And then it looks like regex has a higher priority over word replacement: even though it looks like regex happens after word replacement, it actually happens first? In the picture above, the word "other" should've become "uvvah", but instead it became "othah".
![image](https://github.com/new-frontiers-14/frontier-station-14/assets/65374927/8d05e7b5-1ba4-4db7-808f-fb343c04301f)

Reply:
idk

Reply:
Well, let's call it working, and I'll note it down as a bug.